### PR TITLE
Add handler for making multacc the default SMS app

### PR DESCRIPTION
**Pull request by micah-white**
adds a button in the settings to make Multacc the default sms app. Very fun to write, 10/10 would recommend

### CODE REVIEW

File: lib/pages/settings/settings_page.dart [hunk 1]
```diff
@@ -94,4 +102,37 @@ class _SettingsPageState extends State<SettingsPage> {
...
+  //     final int result = await platform.invokeMethod('getBatteryLevel');
+  //     batteryLevel = 'Battery level at $result % .';
+  //   } on PlatformException catch (e) {
+  //     batteryLevel = "Failed to get battery level: '${e.message}'.";
+  //   }
+
+  //   setState(() {
+  //     _batteryLevel = batteryLevel;
+  //   });
+  // }
```
Comment (mayank99) [inline lib/pages/settings/settings_page.dart:120]: why

File: lib/pages/settings/settings_page.dart [hunk 2]
```diff
@@ -18,6 +19,8 @@ class SettingsPage extends StatefulWidget {
 class _SettingsPageState extends State<SettingsPage> {
   SharedPreferences prefs;
 
+  static const platform = const MethodChannel('com.flutter.sms/default-sms');
```
Comment (mayank99) [inline lib/pages/settings/settings_page.dart:22]:
```suggestion
  static const platform = const MethodChannel('ccom.multacc/sms-handler');
```

File: lib/main.dart [hunk 3]
```diff
@@ -15,7 +15,7 @@ void main() async {
   WidgetsFlutterBinding.ensureInitialized();
 
   // request contacts permission
-  while (!await Permission.contacts.request().isGranted) {}
+  while (!await Permission.contacts.request().isGranted || !await Permission.sms.request().isGranted) {}
```
Comment (mayank99) [inline lib/main.dart:18]: This is not a critical permission, so it needs to be requested separately and only after the default handler is set.

File: android/app/src/main/kotlin/com/example/multacc/MainActivity.kt [hunk 4]
```diff
@@ -1,5 +1,93 @@
...
+  //   if (VERSION.SDK_INT >= VERSION_CODES.LOLLIPOP) {
+  //     val batteryManager = getSystemService(Context.BATTERY_SERVICE) as BatteryManager
+  //     batteryLevel = batteryManager.getIntProperty(BatteryManager.BATTERY_PROPERTY_CAPACITY)
+  //   } else {
+  //     val intent = ContextWrapper(applicationContext).registerReceiver(null, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
+  //     batteryLevel = intent!!.getIntExtra(BatteryManager.EXTRA_LEVEL, -1) * 100 / intent.getIntExtra(BatteryManager.EXTRA_SCALE, -1)
+  //   }
+
+  //   return batteryLevel
+  // }
```
Comment (mayank99) [inline android/app/src/main/kotlin/com/example/multacc/MainActivity.kt:91]: ???

File: android/app/src/main/kotlin/com/example/multacc/MainActivity.kt [hunk 5]
```diff
@@ -1,5 +1,93 @@
...
+import android.os.BatteryManager
+import android.os.Build.VERSION
+import android.os.Build.VERSION_CODES
+import android.provider.Telephony
+
+import android.util.Log
+import io.flutter.plugin.common.PluginRegistry.Registrar
+
+class MainActivity: FlutterActivity() {
+  private val CHANNEL = "com.flutter.sms/default-sms"
```
Comment (mayank99) [inline android/app/src/main/kotlin/com/example/multacc/MainActivity.kt:22]:
```suggestion
  private val CHANNEL = "com.multacc/sms-handler"
```

File: lib/pages/settings/settings_page.dart [hunk 6]
```diff
@@ -94,4 +102,37 @@ class _SettingsPageState extends State<SettingsPage> {
...
+    } catch (e) {
+      print(e);
+    }
+
+    // print(value);
+
+    // setState(() {
+    //   _batteryLevel = batteryLevel;
+    // });
+
```
Comment (mayank99) [inline lib/pages/settings/settings_page.dart:136]: ???

File: lib/pages/settings/settings_page.dart [hunk 7]
```diff
@@ -94,4 +102,37 @@ class _SettingsPageState extends State<SettingsPage> {
...
+  //   setState(() {
+  //     _batteryLevel = batteryLevel;
+  //   });
+  // }
+
+  Future<void> _defaultSMS() async {
+    // int value;
+
+    try {
+      await platform.invokeMethod('defaultSMS');
```
Comment (mayank99) [inline lib/pages/settings/settings_page.dart:126]: This is where you would request the SMS permission

File: lib/pages/settings/settings_page.dart (hunk 1 above)
Comment (micah-white) [reply lib/pages/settings/settings_page.dart:120]: ah yeah forgot to remove those commented stuff

File: android/app/src/main/kotlin/com/example/multacc/MainActivity.kt [hunk 8]
```diff
@@ -1,5 +1,41 @@
...
+import android.util.Log
+import io.flutter.plugin.common.PluginRegistry.Registrar
+
+class MainActivity: FlutterActivity() {
+  private val CHANNEL = "com.multacc/sms-handler"
+
+  override fun configureFlutterEngine(@NonNull flutterEngine: FlutterEngine) {
+    super.configureFlutterEngine(flutterEngine)
+
+	  MethodChannel(flutterEngine.dartExecutor.binaryMessenger, CHANNEL).setMethodCallHandler {
```
Comment (mayank99) [inline android/app/src/main/kotlin/com/example/multacc/MainActivity.kt:27]: This needs to be formatted

File: lib/pages/settings/settings_page.dart [hunk 9]
```diff
@@ -94,4 +102,16 @@ class _SettingsPageState extends State<SettingsPage> {
       prefs.setString('PHONE_APP', value);
     });
   }
+
+
+  Future<void> _defaultSMS() async {
+
+    while(!await Permission.sms.request().isGranted){}
```
Comment (mayank99) [inline lib/pages/settings/settings_page.dart:109]: This is still before the default SMS handler is invoked. It needs to be _after_.

File: android/app/src/main/kotlin/com/example/multacc/MainActivity.kt (hunk 8 above)
Comment (micah-white) [reply android/app/src/main/kotlin/com/example/multacc/MainActivity.kt:27]: In VS it's formatted properly, idk what's going on there

**Pull request merged**